### PR TITLE
feat: move library to typescript code generation

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,32 +25,47 @@
     "Web Risk API"
   ],
   "dependencies": {
-    "google-gax": "^1.7.5"
+    "google-gax": "^1.9.0"
   },
   "devDependencies": {
+    "@types/mocha": "^5.2.5",
+    "@types/node": "^12.0.0",
+    "codecov": "^3.0.0",
```
Comment (JustinBeckwith) [inline package.json]: I'm surprised we need to install codecov here

File: system-test/header.js [hunk 2]
```diff
@@ -1,35 +0,0 @@
-// Copyright 2019 Google LLC
```
Comment (alexander-fenster) [inline system-test/header.js]: Can we have this one back? I think it's useful. Synthtool should not remove it, should it?  We can convert it to TypeScript manually.

File: system-test/test.js [hunk 3]
```diff
@@ -1,30 +0,0 @@
-// Copyright 2019 Google LLC
```
Comment (alexander-fenster) [inline system-test/test.js]: Also this one.

File: webpack.config.js [hunk 4]
```diff
@@ -12,33 +12,51 @@
...
 
+const path = require('path');
+
 module.exports = {
-  entry: './src/browser.js',
+  entry: './src/index.ts',
   output: {
-    library: 'web-risk',
-    filename: './web-risk.js',
+    library: 'WebRiskServiceV1Beta1',
```
Comment (alexander-fenster) [inline webpack.config.js]: So... (1) this is breaking, but (2) what we had before did not work anyway (there must be no dash in the library name), so I guess it's OK to leave it this way.

Comment (xiaozhenliu-gg5) [reply webpack.config.js]: got it, we have package name as `@google-cloud/web-risk` so I guess we may need change from the webpack side?

Comment (xiaozhenliu-gg5) [reply webpack.config.js]: cool

File: system-test/header.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as assert from 'assert';
+import {WebRiskServiceV1Beta1Client} from '../src/v1beta1';
+import * as protoTypes from '../protos/protos';
+
+const http2spy = require('http2spy');
```
[the remark this answers is not in the document]
Comment (xiaozhenliu-gg5) [reply system-test/header.ts]: tried to import it, seems no `@types/http2spy` module available

[the remark this answers is not in the document]
Comment (alexander-fenster) [reply system-test/header.ts]: Oh, that's true. https://github.com/bcoe/http2spy  @bcoe types are badly needed :)

[the remark this answers is not in the document]
Comment (xiaozhenliu-gg5) [reply system-test/header.ts]: yes, I noticed ben's repo from searching http2spy :)

File: package.json (hunk 1 above)
Comment (xiaozhenliu-gg5) [reply package.json]: dependency removed, good call, thanks!

File: system-test/header.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+describe('header', () => {
+  it('populates x-goog-api-client header', async () => {
+    const client = new WebRiskServiceV1Beta1Client();
+    const request = {
+      uri: 'http://testsafebrowsing.appspot.com/s/malware.html',
+      threatTypes: ['MALWARE'],
+    };
+    await client.searchUris(request as unknown as protoTypes.google.cloud.webrisk.v1beta1.SearchUrisRequest);
```
Comment (alexander-fenster) [inline system-test/header.ts]: by the way, why do we cast to `unknown` here?  is the type wrong?

**Comment by alexander-fenster** [comment]
Something is wrong with the header system test:

image

I think you may need to get a service account key and debug it locally :(

File: system-test/header.ts (hunk 6 above)
Comment (xiaozhenliu-gg5) [reply system-test/header.ts]: yea, the two types are not compatible, so the local request lacks `toJSON` method which is not assignable to `google.cloud.webrisk.v1beta1.SearchUriRequest`[toJSON method is required, not optional]

**Comment by xiaozhenliu-gg5** [comment]
cool, system-test fails locally, thought it was because of some credential, but yes, I will get account and debug it. @alexander-fenster

**Pull request merged**